Lib/_pyrepl/reader.py

```diff
@@ -643,6 +645,7 @@ def handle1(self, block: bool = True) -> bool:
             self.dirty = True
 
         while True:
+            self.call_PyOS_InputHook()
```

Comment:
This is a change from the previous behavior. Our input is blocking whereas the old REPL seemed to fire the hook in regular intervals while the prompt was active.

Reply:
This behaviour is undocumented though. This is all we say:

> Can be set to point to a function with the prototype int func(void). The function will be called when Python’s interpreter prompt is about to become idle and wait for user input from the terminal. The return value is ignored. Overriding this hook can be used to integrate the interpreter’s prompt with other event loops, as done in the Modules/_tkinter.c in the Python source code.

The rest is observed behaviour. It would be great to support it exactly as it was but is unclear all the things that people are expecting around it so I am unsure if we can do all of them without compromising everything.

Reply:
I will try a bunch of things before giving the PR another go.

Reply:
> The rest is observed behaviour. It would be great to support it exactly as it was but is unclear all the things that people are expecting around it so I am unsure if we can do all of them without compromising everything.

Speaking as a former matplotlib developer, I think the important thing is that the GUI loop is responsive while also accepting input in the REPL.  I don't know specifically whether that requires `select` ing on both the REPL input and GUI input (as the classic REPL does), or something else would work.  I think just manually testing each of the GUI frameworks is the only way to know.